internal/service/contentprovider/moduleconfig.go

```diff
@@ -94,17 +91,56 @@ type ModuleConfig struct {
 	Annotations         map[string]string          `yaml:"annotations" comment:"optional, additional annotations for the ModuleTemplate"`
 	AssociatedResources []*metav1.GroupVersionKind `yaml:"associatedResources" comment:"optional, GVK of the resources which are associated with the module and have to be deleted with module deletion"`
 	Manager             *Manager                   `yaml:"manager" comment:"optional, the module resource that can be used to indicate the installation readiness of the module. This is typically the manager deployment of the module"`
-	Resources           ResourcesMap               `yaml:"resources,omitempty" comment:"optional, additional resources of the ModuleTemplate that may be fetched"`
+	Resources           Resources                  `yaml:"resources,omitempty" comment:"optional, additional resources of the ModuleTemplate that may be fetched"`
 }
 
-type resource struct {
+type Icons map[string]string
+
+type icon struct {
 	Name string `yaml:"name"`
 	Link string `yaml:"link"`
 }
 
-type ResourcesMap map[string]string
+func (i *Icons) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	icons := []icon{}
```

Comment:
since this is the same logic for UnmarshalYaml for the Resources can we have a common internal function called for both receivers, and maybe unify the struct for both instead of having `icon` and `resource`?

Reply:
🥇 Ah yes, wanted to do that and forgot about it. Thanks, pushed.